### PR TITLE
Update `production` profile documentation

### PR DESCRIPTION
**Pull request by atbrakhi**
In [34340](https://github.com/servo/servo/pull/34340) we optimize `production` profile. This changes updates the book accordingly

### CODE REVIEW

**Comment by atbrakhi** [comment]
> Thanks! I think in this case the details are not too important to mention explicitly, because we have the links. Can we keep the text as is and update the links?

done in [3d99dbb](https://github.com/servo/book/pull/39/commits/3d99dbb3a309c104d57e664d70bbd60253cac15c)

File: src/hacking/building-servo.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ There are three main build profiles, which you can build and use independently o
         <th>optimised?
         <td><a href="https://doc.rust-lang.org/cargo/reference/profiles.html#dev">no</a>
         <td><a href="https://github.com/servo/servo/blob/457d37d94ee6966cad377c373d333a00c637e1ae/Cargo.toml#L153">yes</a>
-        <td>yes, <a href="https://github.com/servo/servo/blob/457d37d94ee6966cad377c373d333a00c637e1ae/Cargo.toml#L159-L160">more</a> <a href="https://github.com/servo/servo/blob/457d37d94ee6966cad377c373d333a00c637e1ae/Cargo.toml#L170-L171">than</a> in <strong>release</strong>
+        <td>yes, <a href="https://github.com/servo/servo/blob/9457a40ca2cd4b9530ba7c5334c82f3b3f2e7ac8/Cargo.toml#L177-L182">more</a> <a href="https://github.com/servo/servo/blob/9457a40ca2cd4b9530ba7c5334c82f3b3f2e7ac8/Cargo.toml#L166-L67">than</a> in <strong>release</strong>
```
Comment (delan) [inline src/hacking/building-servo.md:60]:
```suggestion
        <td>yes, <a href="https://github.com/servo/servo/blob/9457a40ca2cd4b9530ba7c5334c82f3b3f2e7ac8/Cargo.toml#L177-L182">more than</a> in <strong>release</strong>
```

Comment (delan) [reply src/hacking/building-servo.md:60]: (the second link seems to be broken anyway)

**Review by delan — APPROVED** [review]
Great, thanks!

**Pull request merged**